From ef9e3030d556b2b0ca2bb85fd4087497f7538d4c Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Tue, 9 Nov 2010 10:16:56 +0000
Patch-mainline: 2.6.38
References: fate#310916
Git-commit: e74cfed521746544e6eeee84b24bd31c1e59ffe2
Subject: [PATCH 1856/2588] Revert "drm/i915/ringbuffer: Ignore failure to
 setup the ring on Sandybridge"

This reverts commit 629e894173c9de589913cf649deaadec4b0579bd.
(cherry picked from commit e74cfed521746544e6eeee84b24bd31c1e59ffe2)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_ringbuffer.c |   32 +++++++-----------------------
 1 files changed, 8 insertions(+), 24 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_ringbuffer.c b/drivers/gpu/drm/i915/intel_ringbuffer.c
index 8728c1e..5ea6ad3 100644
--- a/drivers/gpu/drm/i915/intel_ringbuffer.c
+++ b/drivers/gpu/drm/i915/intel_ringbuffer.c
@@ -182,30 +182,14 @@ static int init_ring_common(struct intel_ring_buffer *ring)
 	if ((I915_READ_CTL(ring) & RING_VALID) == 0 ||
 	    I915_READ_START(ring) != obj_priv->gtt_offset ||
 	    (I915_READ_HEAD(ring) & HEAD_ADDR) != 0) {
-		if (IS_GEN6(ring->dev) && ring->dev->pdev->revision <= 8) {
-			/* Early revisions of Sandybridge do not like
-			 * revealing the contents of the ring buffer
-			 * registers whilst idle. Fortunately, the
-			 * auto-reporting mechanism prevents most hangs,
-			 * but this will bite us eventually...
-			 */
-			DRM_DEBUG("%s initialization failed "
-				  "ctl %08x head %08x tail %08x start %08x. Ignoring, hope for the best!\n",
-				  ring->name,
-				  I915_READ_CTL(ring),
-				  I915_READ_HEAD(ring),
-				  I915_READ_TAIL(ring),
-				  I915_READ_START(ring));
-		} else {
-			DRM_ERROR("%s initialization failed "
-				  "ctl %08x head %08x tail %08x start %08x\n",
-				  ring->name,
-				  I915_READ_CTL(ring),
-				  I915_READ_HEAD(ring),
-				  I915_READ_TAIL(ring),
-				  I915_READ_START(ring));
-			return -EIO;
-		}
+		DRM_ERROR("%s initialization failed "
+				"ctl %08x head %08x tail %08x start %08x\n",
+				ring->name,
+				I915_READ_CTL(ring),
+				I915_READ_HEAD(ring),
+				I915_READ_TAIL(ring),
+				I915_READ_START(ring));
+		return -EIO;
 	}
 
 	if (!drm_core_check_feature(ring->dev, DRIVER_MODESET))
-- 
1.7.6

